### PR TITLE
feat(1284): implemented api-catalog-spa beta

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Closes/Fixes/Resolves

Closes 1284

# Explain the feature/fix

1. Implemented API Catalog SPA CRUD operations
2. Implemented catalog spa API toolset view
3. Subscription option and search

## Does this PR introduce a breaking change

No

## Screenshots

<details>
<summary>View Screenshots</summary>

## Home Page

Screenshot 2021-11-03 at 5 13 01 PM

## Explore Page

Screenshot 2021-11-03 at 5 13 19 PM

## API Add Page

Screenshot 2021-11-03 at 5 13 10 PM

## API Details Page
Screenshot 2021-11-03 at 5 13 34 PM

## API Details Page with subscription

Screenshot 2021-11-03 at 5 14 56 PM

## API Edit Page

Screenshot 2021-11-03 at 5 15 08 PM

## API Delete Confirmation

Screenshot 2021-11-03 at 5 15 14 PM

## GQL Playground

Screenshot 2021-11-03 at 5 15 26 PM

## Voyager

Screenshot 2021-11-03 at 5 16 58 PM

## API Detail Page - Rest

Screenshot 2021-11-03 at 5 17 59 PM

## API Swagger Page

Screenshot 2021-11-03 at 5 18 07 PM

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { Split, SplitItem, FormGroup, FormAlert, Alert } from '@patternfly/react-core';
+import { Controller, useFormContext } from 'react-hook-form';
+import { callbackify } from 'utils';
+import { CatalogBigButton } from 'components/CatalogBigButton';
+import { FormData } from '../../types';
+
+const apiOptions = [
+  {
+    title: 'REST',
+    desc: 'Conforms to OpenAPI specs',
```
Comment (riginoommen) [inline packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx:10]:
```suggestion
    desc: 'Confirms to OpenAPI specs',
```

File: packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    image: 'rest-logo.svg',
+    type: 'REST',
+  },
+  {
+    title: 'Graphql',
```
Comment (riginoommen) [inline packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx:15]:
```suggestion
    title: 'GraphQL',
```

**Review by riginoommen — CHANGES_REQUESTED** [review]
Need changes as mentioned in above comments also 

Looks like packages/api-catalog-spa/src/router/links.ts links need a change

File: packages/api-catalog-spa/src/router/links.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export enum ApiCatalogLinks {
+  HomePage = '/',
+  ListPage = '/apis',
+  AddNewApiPage = '/apis/new',
+  ApiDescriptionPage = '/apis/:id',
+  ApiEditPage = '/apis/:id/edit',
```
Comment (riginoommen) [inline packages/api-catalog-spa/src/router/links.ts:6]:
```suggestion
  ApiEditPage = '/apis/:id/edit',
```

WIth the slug the api should do the operations. mongodb id no need to to show it

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx (hunk 1 above)
Comment (akhilmhdh) [reply packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx]: Fixed

File: packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx (hunk 2 above)
Comment (akhilmhdh) [reply packages/api-catalog-spa/src/pages/ApiCUDPage/components/ApiTypeSelector/ApiTypeSelector.tsx]: Done

File: packages/api-catalog-spa/src/router/links.ts (hunk 3 above)
Comment (akhilmhdh) [reply packages/api-catalog-spa/src/router/links.ts:6]: @riginoommen That id and mongodb id has no relation it for frontend id positioning. But yah Mongo id is used as we discussed. 

I have added nanoid(6) as slug  for now

**Review by riginoommen — APPROVED** [review]

**Review by deshmukhmayur — APPROVED** [review]

**Pull request merged**